pkg/cloud/cloud.go

```diff
@@ -18,26 +18,30 @@ limitations under the License.
 
 package cloud
 
-import (
-	"github.com/spf13/cobra"
-	"github.com/submariner-io/submariner-operator/internal/restconfig"
-	"github.com/submariner-io/submariner-operator/pkg/subctl/cmd"
-	"github.com/submariner-io/submariner-operator/pkg/subctl/cmd/cloud/cleanup"
-	"github.com/submariner-io/submariner-operator/pkg/subctl/cmd/cloud/prepare"
-)
+type Port struct {
+	Natt         uint16
+	NatDiscovery uint16
+	VxLAN        uint16
+	Metrics      uint16
+}
 
-var (
-	cloudCmd = &cobra.Command{
-		Use:   "cloud",
-		Short: "Cloud operations",
-		Long:  `This command contains cloud operations related to Submariner installation.`,
-	}
-	restConfigProducer = restconfig.NewProducer()
-)
+type Instance struct {
+	AWSGWType string
+	GCPGWType string
```

Comment:
I assume the purpose here is to reuse a struct between different cloud platform. That's fine but we shouldn't have provider-specific fields. Either combine into a generic field, eg `GatewayType`, or define structs for each platform.